plugins/cairo-lang-quote/src/lib.rs

```diff
@@ -108,8 +119,84 @@ pub fn quote(input: RustTokenStream) -> RustTokenStream {
             }),
         }
     }
-    RustTokenStream::from(rust_quote!({
+    proc_macro::TokenStream::from(rust_quote!({
       #output_token_stream
       quote_macro_result
     }))
 }
+
+#[cfg(test)]
+mod tests {
+    use super::{process_token_stream, QuoteToken};
+    use proc_macro2::{Ident, Span};
+    use quote::{quote as rust_quote, TokenStreamExt};
+
+    #[test]
+    fn parse_cairo_attr() {
+        let input: proc_macro2::TokenStream = rust_quote! {
+            #[some_attr]
+            fn some_fun() {
+
+            }
+        };
+        let mut output = Vec::new();
+        process_token_stream(input.into_iter().peekable(), &mut output);
+        assert_eq!(
+            output,
+            vec![
+                QuoteToken::Content("#".to_string()),
+                QuoteToken::Content("[".to_string()),
+                QuoteToken::Content("some_attr".to_string()),
+                QuoteToken::Content("]".to_string()),
+                QuoteToken::Content("fn".to_string()),
+                QuoteToken::Whitespace,
+                QuoteToken::Content("some_fun".to_string()),
+                QuoteToken::Content("(".to_string()),
+                QuoteToken::Content(")".to_string()),
+                QuoteToken::Content("{".to_string()),
+                QuoteToken::Content("}".to_string()),
+            ]
+        );
+    }
+
+    #[test]
+    fn quote_var_whitespace() {
```

Comment:
Before this fix, `mod #name {` got translated to ` modname {` 🙃

Reply:
That's on me, my bad
<img src="https://github.com/user-attachments/assets/69821580-4824-4eaa-8dd5-886fd75daf26" width="200" height="150" />